avocado/export/_base.py

```diff
@@ -90,11 +193,9 @@ def read(self, iterable, force_distinct=True, offset=None, limit=None,
 
             if offset is None or i >= offset:
                 emitted += 1
-                yield self._format_row(_row, **kwargs)
+
+                yield self._format_row(_row, kwargs=kwargs)
 
     def write(self, iterable, *args, **kwargs):
-        for row_gen in self.read(iterable, *args, **kwargs):
```

Comment:
@murphyke Another change to support alternate methods of creating the formatting iterable is *not* having the `Exporter.write` method call `read`. The iterable is expected to produce rows that are already formatted.